### ISSUE TITLE
Build errors with boost headers with clang since 13

### ISSUE THREAD

**Issue by LABPBS** [msys2/MINGW-packages#9827]
```
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/ip/tcp.hpp:19:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/basic_socket_acceptor.hpp:19:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/any_io_executor.hpp:22:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution.hpp:18:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution/allocator.hpp:20:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution/executor.hpp:20:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution/execute.hpp:20:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution/detail/as_invocable.hpp:20:
D:/a/_temp/msys64/clang64/include/boost/asio/detail/memory.hpp:88:20: error: reference to unresolved using declaration
  void* ptr = std::aligned_alloc(align, size);
                     ^
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/ip/tcp.hpp:19:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/basic_socket_acceptor.hpp:19:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/any_io_executor.hpp:22:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution.hpp:18:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution/allocator.hpp:20:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution/executor.hpp:20:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution/execute.hpp:20:
In file included from D:/a/_temp/msys64/clang64/include/boost/asio/execution/detail/as_invocable.hpp:20:
D:/a/_temp/msys64/clang64/include/boost/asio/detail/memory.hpp:88:15: error: excess elements in scalar initializer
  void* ptr = std::aligned_alloc(align, size);
              ^                       ~~~~~~
```
_Originally posted by @LABPBS in https://github.com/msys2/MINGW-packages/pull/9823#issuecomment-947841188_

**Comment by Biswa96** [comment]
> Boost ASIO says... libc++ says...

Little confusion. How did you decide who is wrong here?

**Comment by jeremyd2019** [comment]
This seems to have been fixed in chriskohlhoff/asio@92aacba.  This patch should be able to be applied to boost.

**Comment by oscarfv** [comment]
> This seems to have been fixed in [chriskohlhoff/asio@92aacba](https://github.com/chriskohlhoff/asio/commit/92aacba). This patch should be able to be applied to boost.

FWIW: the patch is included on the next boost release, which is planned to be released on December 8.

**Comment by oscarfv** [comment]
There are more problems of this type, as for example clang+libc++ needing `BOOST_ASIO_HAS_STD_INVOKE_RESULT` on `-std=c++20`, which is also fixed in the upcoming release.

It is quite common to find problems when upgrading to a compiler that was released after boost. IMHO the best course of action is to wait for a new boost release and add local workarounds in the meanwhile.
